lib/rules/i18n-no-reference.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const { I18N_METHODS } = require('../i18n.js');
+
+function isReferenceNode(node) {
+  if (node.type === 'TemplateLiteral') return false;
+
+  return node.type !== 'Literal' || typeof node.value !== 'string'
+}
+
+function extractI18nMethodAndArgs(node) {
+  let result = [];
+
+  if (node.type === 'CallExpression') {
+    const { callee } = node;
+    if (callee.type === 'MemberExpression') {
+      const { object, property } = callee;
+      if (object.name === 'i18n') {
+        result = [property.name, node.arguments];
+      }
+    }
+  }
+
+  return result;
+}
+
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Possible Errors',
+      description:
+        'Ensure predictable static values are passed as i18n method arguments',
+      recommended: false,
+    },
+    fixable: null, // Indicates that some problems reported by this rule are automatically fixable
+    schema: [],
+    messages: {
+      noDynamicValue:
+        'invalid argument type "{{ arg }} passed to i18n.{{ method }}()".',
+      noDynamicValueForSprintf:
+        'invalid argument type "{{ arg }}" passed to i18n.sprintf(). Only static values or calls to i18n methods are allowed',
+    },
+  },
+
+  create(context) {
+    return {
+      CallExpression(node) {
+        const [methodName, methodArgs] = extractI18nMethodAndArgs(node);
+
+        if (!methodName || !methodArgs) return;
+
+        if (methodName.includes('ngettext')) return;
+
+        if (methodName === 'sprintf') {
+          const [firstArg] = methodArgs;
+
+          if (isReferenceNode(firstArg)) {
+            const [argMethod] = extractI18nMethodAndArgs(firstArg);
+
+            if (!argMethod) {
+              context.report({
+                node,
+                messageId: 'noDynamicValueForSprintf',
+                data: {
+                  arg: firstArg.type,
+                  method: methodName,
+                },
+              });
```

Comment:
shouldn't we early return here?

Reply:
Looking at it again, it actually makes more sense to add sprintf with positions [0] to the i18n config. It is totally valid for the other rules and means we don't even need this conditional at all. wdyt?

Reply:
sounds good